leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/persistence/ExpirationChecker.java

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Alexander Ellwein, Daniel Maier (Bosch Software Innovations GmbH)
+ *                                - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.server.queue.persistence;
+
+import java.util.Calendar;
+
+/**
+ * Expiration checker is responsible to check expiration of a particular queue request (entity). An implementation of an
+ * expiration checker may be specific to a particular QueuePersistence implementation.
+ */
+public interface ExpirationChecker {
```

Comment:
I agree with @jvermillard about expiration.
Since expiration is not mentioned in the specification, I'm not sure it is needed in the API. 
From what I understand of the Queue Mode mechanism, requests are not expired but I assume they must be cleaned on client de-registration.
I think that expiration should be moved to the implementation. Does it make sense?